pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -83,24 +87,39 @@ type stateData struct {
 	// (if one exists) and the changes made to it.
 	podSchedulingState podSchedulingState
 
+	// resourceModel contains the information about available and allocated resources when using
+	// structured parameters and the pod needs this information.
+	resources resources
+
+	// mutex must be locked while accessing any of the fields below.
 	mutex sync.Mutex
 
 	informationsForClaim []informationForClaim
 }
 
+func (d *stateData) Clone() framework.StateData {
```

Comment:
Am I correct to assume that at this point there is no support for preemption?

Reply:
Correct.

With structured parameters there's probably enough information to implement it, but for now let's keep it as simple as possible.

We also need to think about scoring, which is probably even more important. There is no way how the scheduler can tell that one selection is "better" than some other.